### ISSUE TITLE
FBNEO renamer not working for games that have "sampleof=" in dat file entry

### ISSUE THREAD

**Issue by OldFinbar** [gerrykeys/playnite-mame-utility#13]
If a game entry in the FBNEO dat file has the string "sampleof=[rom name]" it will not rename the file.

For example:

\<game name="buckrogn" cloneof="buckrog" romof="buckrog" sampleof="buckrog" sourcefile="sega/d_turbo.cpp">
\<description>Buck Rogers: Planet of Zoom (not encrypted, set 1)\</description>

will not be renamed. However, if the **_sampleof="buckrog"_** string is removed, the file will be renamed to **Buck Rogers: Planet of Zoom (not encrypted, set 1)** as it is supposed to be.

Many thanks for this great extension.

**Comment by gerrykeys** [comment]
Hello, thanks for report this issue. I will fix this asap

**Comment by gerrykeys** [comment]
Hello,
sorry for the long delay.

I looked for the issue and I'm not sure that a rom can be considered a game if it is marked as "sampleof" inside the xml/dat file.
If you have more specific information about that please tell me.

Anyway i will release soon a new version of plugin where if a rom is marked as "sampleof" and it is marked also as "cloneof" it will be considered a Game.

So, for the moment i will close this ticket as "not a bug".

**Comment by OldFinbar** [comment]
Hi gerrykeys, no problem regarding the delay. 

The "sampleof" is referring to the audio samples that the game uses that are required for some games to have proper audio. So the clone version of Buck Rogers (buckrogn), it requires the audio samples from the original version of Buck Rogers (buckrog) and that is what the "sampleof" is referencing.

**Comment by gerrykeys** [comment]
Ok i think my fix in the latest version 1.0.6 should fix your issue.
Let me know if it is ok.
Thanks

**Comment by OldFinbar** [comment]
Hi gerrykeys,

I have tested version 1.0.6 and now it is only missing renaming when the game is marked with "sampleof" but not "cloneof".

What this means is that parent roms that require an audio sample using "sampleof" will not be renamed, but clone games which use "sampleof" will be renamed. 

Some examples of games that will not be renamed are Hustle (hustle) and CoMotion (comotion) which both use audio samples from Blockade (sampleof=blockade).

However, the clone version of CoMotion which is called "comotionp" will be renamed.

Please let me know if you need any further details as I am happy to help.

**Comment by gerrykeys** [comment]
Hi @OldFinbar 
thanks for reported that issue:

the point is the following:
Are we sure that all the entry machines inside the xml/dat file tagged with only "Sample of" are ALL games?
If this is correct then i will fix the plugin.

**Comment by OldFinbar** [comment]
> the point is the following: Are we sure that all the entry machines inside the xml/dat file tagged with only "Sample of" are ALL games?

Yes, I have checked using the latest version of FinalBurn Neo dat file. There are 166 total hits for "sampleof". 20 are parent roms and I tested all of them to check that they are games. The remaining 146 hits are "cloneof=" and I have verified that they are all games.

**Comment by gerrykeys** [comment]
Hi @OldFinbar 

can you take a look inside [this](http://adb.arcadeitalia.net/download_file.php?tipo=emulator&id=998&codice=gamelist.xml.zip&rename=mame271-gamelist.xml.zip) xml gamelist file?

It comes from Mame application.
If your analysis is good also for that file than i will do the fix.

Thanks for your support, it is very precious.

**Comment by OldFinbar** [comment]
You are welcome for the support and thank you for the work you have put into this great extension.

I have looked through the MAME xml and noted that only the mechanical / device / bios tags are necessary to determine whether the machine is a standard game or not. There are some mechanical / device / bios machines that use samples, however these will already be tagged as non-games/mechanical by your extension I believe. For example: \<machine name="allied" sourcefile="pinball/allied.cpp" isbios="yes" sampleof="genpin"> uses samples, but it should already be tagged as a bios machine.

The main website that people use to download MAME audio samples is https://www.progettosnaps.net/samples/ and it contains a good explanation of what the sample files and "sampleof" tag means:

> Some sets (games) of MAME need some additional files to emulate the audio perfectly, these files are contained in compressed zip package with the same name of the file containing the roms and it should be placed in the "samples" folder of your MAME .

Basically, whether a machine requires samples (sampleof=) is irrelevant to whether it is a game or not. The "sampleof" tag only shows whether these extra audio files are required or not.

**Comment by gerrykeys** [comment]
@OldFinbar 
Thanks a lot for your analysis!
So i can finally fix the issue.
I will let you know when i release the new version (i think today/tomorrow), so you can test it.
Thanks again

**Comment by gerrykeys** [comment]
Hey @OldFinbar 

i just released a new version 1.0.7 with the fix.
I also thanked you in the release notes.

Let me know it works for you.
Thanks again.

**Comment by OldFinbar** [comment]
Hello @gerrykeys 

I have tested 1.0.7 with both FBNeo xml and using MAME executable and it works very well! Thank you very much for this. 
All games are now renamed, including those that use samples. Bios and devices are skipped when renaming.
I tested all of the removing options and they all work perfectly.

Fantastic, thank you again! Your extension makes Playnite handle MAME and FBNeo wonderfully. I love the tagging feature as well.
